application/extensions/DateTimePickerWidget/DateTimePicker.php

```diff
@@ -198,8 +198,15 @@ private function getLocalizationOptionsString()
         }
         $localeScript .= "      dayViewHeaderFormat: { month: 'long', year: 'numeric' },\n" .
             "      locale: '$locale',\n" .
-            "      format: '$dateFormat',\n" .
-            "      hourCycle: 'h24'\n";
+            "      format: '$dateFormat',\n";
+
+        // Try to guess the right hour cycle from the format
+        // The old datetimepicker did that, but Tempus Dominus does not (it guesses from the locale, not the format)
+        if (strpos(strtolower($dateFormat), 'a') === false && strpos($dateFormat, 'h') === false) {
```

Comment:
Why "h" ? "HH:MM A" for example ?

https://getdatepicker.com/6/plugins/customDateFormat.html

Reply:
We picked it up from LSv3.

> Why "h" ?
Stands for `1-2 digit hour (12 hour)`. I guess that signals to expect a 12 hour cycle.

Reply:
HH is already for 24hour format.
What kind of automatic test do you picture?

Reply:
> HH is already for 24hour format.

right :) sorry

>  What kind of automatic test do you picture?

Using different time format in 2 or 3 questions, with default answers set to something in afternoon.
And check if value shown is OK ?